docs/howto/logging.txt

```diff
@@ -0,0 +1,336 @@
+to organize log messages according to their provenance within your project's
+applications automatically. It also ensures that you will not experience name
+collisions.
+
+A logger mapping named ``my_app.views`` will capture records from this logger:
+
+..  code-block:: python
+    :emphasize-lines: 4
+
+    LOGGING = {
+        [...]
+        'loggers': {
+            'my_app.views': {
+                ...
+            },
+        },
+    }
+
+A logger mapping named ``my_app`` will be more permissive, capturing records
+from loggers anywhere within the ``my_app`` namespace (including
+``my_app.views``, ``my_app.utils``, and so on):
+
+..  code-block:: python
+    :emphasize-lines: 4
+
+    LOGGING = {
+        [...]
+        'loggers': {
+            'my_app': {
+                ...
+            },
+        },
+    }
+
+You can also define logger namespacing explicitly::
+
+    logger = logging.getLogger('project.payment')
+
+and set up logger mappings accordingly.
+
+.. _naming-loggers-hierarchy:
+
+Using logger hierarchies and propagation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Logger naming is *hierarchical*. ``my_app`` is the parent of ``my_app.views``,
+which is the parent of ``my_app.views.private``. Unless specified otherwise,
+logger mappings will propagate the records they process to their parents - a
+record from a logger in the ``my_app.views.private`` namespace will be handled
+by a mapping for both ``my_app`` and ``my_app.views``.
+
+To manage this behavior, set the propagation key on the mappings you define::
+
+    LOGGING = {
+        [...]
+        'loggers': {
+            'my_app': {
+                [...]
+            },
+            'my_app.views': {
+                [...]
+            },
+            'my_app.views.private': {
+                [...]
+                'propagate': False,
+            },
+        },
+    }
+
+``propagate`` defaults to ``True``; in this example, the logs from
+``my_app.views.private`` will not be handled by the parent, but logs from
+``my_app.views`` will.
+
+Configure responsive logging
+----------------------------
+
+Logging is most useful when it contains as much information as possible, but
+not information that you don't need - and how much you need depends upon what
+you're doing. When you're debugging, you need a level of information that would
+be excessive and unhelpful if you had to deal with it in production.
+
+You can configure logging to provide you with the level of detail you need,
+when you need it. Rather than manually change configuration to achieve this, a
+better way is to apply configuration automatically according to the environment.
+
+For example, you could set an environment variable ``DJANGO_LOG_LEVEL``
+appropriately in your development and QA environments, and make use of it in a
+logger mapping thus::
+
+    'level': os.getenv('DJANGO_LOG_LEVEL', 'WARNING')
+
+\- so that unless the environment specifies a lower log level, this
```

Comment:
Do we need `\-`? :thinking:

Reply:
Yes, it needs to be escaped otherwise the `-` is interpreted as list item. 

(Or are you are questioning the `-`? I feel that's appropriate there, to introduce a clear pause in the sentence between the action and the reason for it.)

Reply:
I think it's OK as is. Starting the new line with the hyphen to indicate the sentence continuation is allowed I think. 

Wondering is the paragraph couldn't do with another take **but** remembering the diátaxis method, I don't think we should block on it. Let's get this in, and we can follow-up with an improvement if one comes up. 

(I tried sitting down with it, but I had my second jab yesterday, and my head wouldn't work...)

Reply:
This is a precedent in our docs, but OK we can leave it.